google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -1391,6 +1407,24 @@ public Builder setEnableExtendedTracing(boolean enableExtendedTracing) {
       return this;
     }
 
+    /**
+     * Enable spanner server side tracing. Enabling this option will create the trace spans at the
+     * Spanner layer. By default, server side tracing is disabled.
+     */
+    public Builder enableServerSideTracing() {
```

Comment:
Given that this is a configuration option set once (per SpannerOptions) and doesn't change during runtime, the current api looks more natural to me. I noticed other similar usages for enableOpenTelemetryTraces, DirectPath, LAR, and GrpcGcpExtension. Is there any specific reason for choosing the set* approach for API tracing? If not, I would prefer to keep the enable/disable API.

Reply:
Because it is much easier for any application that wants to configure this based on external configuration to use a setter instead of separate enable/disable methods. If for example an application wants to enable this based on a value that is read from a configuration file, they can just create their code like this:

```
SpannerOptions options = SpannerOptions.newBuilder()
   .setProject("my-project")
   .setOtherConfigOption("some-value")
   .setEnableServerTracing(valueFromConfigFile)
   .build;
```

However, with the enable/disable strategy, the code looks like this:

```
SpannerOptions.Builder builder = SpannerOptions.newBuilder()
   .setProject("my-project")
   .setOtherConfigOption("some-value");
if (valueFromConfigFile) {
   builder.enableServerTracing();
}
SpannerOptions options = builder.build();
```

If we then have a lot of enable/disable type of methods, then it gets even more verbose.

Reply:
Thanks Knut. I can see how people might be using configuration files to init SpannerOptions particularly for things like project_id. Using Set* methods sounds good to me.

Reply:
Done. Combined enable/disable into a single set function. Thanks Knut.